diff --git a/client/main.html b/client/main.html
index 09a8a2a..5136977 100755
--- a/client/main.html
+++ b/client/main.html
@@ -32,7 +32,7 @@ with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
       height: 100%;
       font-family: 'Source Sans Pro', Arial, sans-serif;
       font-size: 1rem; /* 16px */
-      background-color: #06172A;
+      background-color: #669999;
     }
 
     a {
diff --git a/client/stylesheets/bbb-icons.css b/client/stylesheets/bbb-icons.css
index 901692a..9b97245 100755
--- a/client/stylesheets/bbb-icons.css
+++ b/client/stylesheets/bbb-icons.css
@@ -32,7 +32,7 @@
   content: "\e932";
 }
 .icon-bbb-unmute:before {
-  content: "\e931";
+  content: "\e962";
 }
 .icon-bbb-file:before {
   content: "\e92e";
@@ -112,7 +112,7 @@
   content: "\e904";
 }
 .icon-bbb-user:before {
-  content: "\e905";
+  content: "\e960";
 }
 .icon-bbb-up_arrow:before {
   content: "\e906";
@@ -124,7 +124,7 @@
   content: "\e90b";
 }
 .icon-bbb-listen:before {
-  content: "\e90c";
+  content: "\e961";
 }
 .icon-bbb-left_arrow:before {
   content: "\e90d";
@@ -220,10 +220,10 @@
   content: "\e93c";
 }
 .icon-bbb-multi_whiteboard:before {
-  content: "\e93d";
+  content: "\e966";
 }
 .icon-bbb-whiteboard:before {
-  content: "\e93e";
+  content: "\e905";
 }
 .icon-bbb-rooms:before {
   content: "\e93f";
diff --git a/imports/api/guest-users/server/handlers/guestsWaitingForApproval.js b/imports/api/guest-users/server/handlers/guestsWaitingForApproval.js
index 5f9618c..5176c4e 100644
--- a/imports/api/guest-users/server/handlers/guestsWaitingForApproval.js
+++ b/imports/api/guest-users/server/handlers/guestsWaitingForApproval.js
@@ -5,7 +5,7 @@ import GuestUsers from '/imports/api/guest-users/';
 
 const COLOR_LIST = [
   '#7b1fa2', '#6a1b9a', '#4a148c', '#5e35b1', '#512da8', '#4527a0',
-  '#311b92', '#3949ab', '#303f9f', '#283593', '#1a237e', '#1976d2', '#1565c0',
+  '#311b92', '#3949ab', '#303f9f', '#283593', '#1a237e', '#1976d2', '#669999',
   '#0d47a1', '#0277bd', '#01579b',
 ];
 
diff --git a/imports/api/users/server/modifiers/addUser.js b/imports/api/users/server/modifiers/addUser.js
index 6229dee..57c56a3 100755
--- a/imports/api/users/server/modifiers/addUser.js
+++ b/imports/api/users/server/modifiers/addUser.js
@@ -12,9 +12,9 @@ import flat from 'flat';
 import addVoiceUser from '/imports/api/voice-users/server/modifiers/addVoiceUser';
 
 const COLOR_LIST = [
-  '#7b1fa2', '#6a1b9a', '#4a148c', '#5e35b1', '#512da8', '#4527a0',
-  '#311b92', '#3949ab', '#303f9f', '#283593', '#1a237e', '#1976d2', '#1565c0',
-  '#0d47a1', '#0277bd', '#01579b',
+  '#7b1fa2', '#6a1b9a', '#4a148c', '#5e35b1', '#512da8', '#98f4a6',
+  '#311b92', '#3949ab', '#303f9f', '#669999', '#1a237e', '#1976d2', '#1565c0',
+  '#669999', '#0277bd', '#01579b',
 ];
 
 export default function addUser(meetingId, userData) {
diff --git a/imports/ui/components/button/styles.scss b/imports/ui/components/button/styles.scss
index ce751fa..2146750 100755
--- a/imports/ui/components/button/styles.scss
+++ b/imports/ui/components/button/styles.scss
@@ -127,7 +127,7 @@
   &:active {
     &:focus {
       span:first-of-type::before {
-        border-radius: 50%;
+        border-radius: 5%;
         @extend %highContrastOutline;
         outline-style: solid;
       }
diff --git a/imports/ui/components/nav-bar/component.jsx b/imports/ui/components/nav-bar/component.jsx
index fd4887a..130b704 100755
--- a/imports/ui/components/nav-bar/component.jsx
+++ b/imports/ui/components/nav-bar/component.jsx
@@ -15,6 +15,8 @@ import ConnectionStatusButton from '/imports/ui/components/connection-status/but
 import ConnectionStatusService from '/imports/ui/components/connection-status/service';
 import SettingsDropdownContainer from './settings-dropdown/container';
 
+import StreamingIndicator from './streaming-indicator/container';
+
 const intlMessages = defineMessages({
   toggleUserListLabel: {
     id: 'app.navBar.userListToggleBtnLabel',
@@ -125,6 +127,10 @@ class NavBar extends Component {
               mountModal={mountModal}
               amIModerator={amIModerator}
             />
+            <StreamingIndicator
+              mountModal={mountModal}
+              amIModerator={amIModerator}
+            />
           </div>
           <div className={styles.right}>
             {ConnectionStatusService.isEnabled() ? <ConnectionStatusButton /> : null}
diff --git a/imports/ui/components/settings/submenus/application/component.jsx b/imports/ui/components/settings/submenus/application/component.jsx
index d549e5a..507fe1f 100644
--- a/imports/ui/components/settings/submenus/application/component.jsx
+++ b/imports/ui/components/settings/submenus/application/component.jsx
@@ -334,6 +334,8 @@ class ApplicationMenu extends BaseMenu {
           {this.renderAudioFilters()}
           {this.renderPaginationToggle()}
 
+          <hr className={styles.separator} />
+
           <div className={styles.row}>
             <div className={styles.col} aria-hidden="true">
               <div className={styles.formElement}>
@@ -370,7 +372,6 @@ class ApplicationMenu extends BaseMenu {
             </div>
           </div>
 
-          <hr className={styles.separator} />
           <div className={styles.row}>
             <div className={styles.col}>
               <div className={styles.formElement}>
diff --git a/imports/ui/components/status-notifier/component.jsx b/imports/ui/components/status-notifier/component.jsx
index eea6223..4c484e4 100644
--- a/imports/ui/components/status-notifier/component.jsx
+++ b/imports/ui/components/status-notifier/component.jsx
@@ -30,7 +30,12 @@ const messages = defineMessages({
     id: 'app.statusNotifier.and',
     description: 'used as conjunction word',
   },
+  lowerHandsOkLabel: {
+    id: 'app.meeting.endNotification.ok.label',
+    description: 'text displayed to clear all raised hands',
+  },
 });
+// app.about.confirmLabel
 
 const MAX_AVATAR_COUNT = 3;
 
@@ -176,7 +181,7 @@ class StatusNotifier extends Component {
         <div className={toastStyles.separator} />
         <Button
           className={styles.clearBtn}
-          label={intl.formatMessage(messages.lowerHandsLabel)}
+          label={intl.formatMessage(messages.lowerHandsOkLabel)}
           color="default"
           size="md"
           onClick={() => {
diff --git a/imports/ui/components/user-avatar/styles.scss b/imports/ui/components/user-avatar/styles.scss
index ffb7785..d7da535 100755
--- a/imports/ui/components/user-avatar/styles.scss
+++ b/imports/ui/components/user-avatar/styles.scss
@@ -106,7 +106,7 @@
 }
 
 .moderator {
-  border-radius: 5px;
+  border-radius: 30% 70% 70% 30% / 30% 30% 70% 70%;
 }
 
 .presenter {
@@ -144,7 +144,7 @@
 .noVoice {
   &:after {
     content: "";
-    background-color: var(--color-off-white);
+    background-color: var(--color-danger);
     top: 1.375rem;
     left: 1.375rem;
     right: auto;
diff --git a/imports/ui/stylesheets/variables/palette.scss b/imports/ui/stylesheets/variables/palette.scss
index e1c12e7..70efea9 100644
--- a/imports/ui/stylesheets/variables/palette.scss
+++ b/imports/ui/stylesheets/variables/palette.scss
@@ -4,7 +4,7 @@
 
   --color-black: #000000;
   --color-gray: #4E5A66;
-  --color-gray-dark: #06172A;
+  --color-gray-dark: #669999;
   --color-gray-light: #8B9AA8;
   --color-gray-darkest: #111111;
   --color-gray-lighter: #a7b3bd;
@@ -14,7 +14,7 @@
   --color-blue-lighter: #92BCEA;
   --color-blue-lightest: #E4ECF2;
 
-  --color-primary: #0F70D7;
+  --color-primary: #38828f;
   --color-success: #008081;
   --color-danger: #DF2721;
   --color-warning: purple;
diff --git a/public/locales/en.json b/public/locales/en.json
index a4887cb..3810b34 100755
--- a/public/locales/en.json
+++ b/public/locales/en.json
@@ -800,5 +800,14 @@
     "app.debugWindow.form.userAgentLabel": "User Agent",
     "app.debugWindow.form.button.copy": "Copy",
     "app.debugWindow.form.enableAutoarrangeLayoutLabel": "Enable Auto Arrange Layout",
-    "app.debugWindow.form.enableAutoarrangeLayoutDescription": "(it will be disabled if you drag or resize the webcams area)"
+    "app.debugWindow.form.enableAutoarrangeLayoutDescription": "(it will be disabled if you drag or resize the webcams area)",
+
+    "app.notification.streamingStart": "Start",
+    "app.notification.streamingStop": "Stop",
+    "app.notification.streamingAriaLabel": "streamingAriaLabel",
+    "app.streaming.startTitle": "startTitle",
+    "app.streaming.stopTitle": "stopTitle",
+    "app.streaming.resumeTitle": "resumeTitle",
+    "app.navBar.streaming.on": "On",
+    "app.navBar.streaming.off": "Off"
 }
